### DISCUSSION TITLE
Inpainting way too sharp, bug or not?

### DISCUSSION THREAD

**Discussion by xunkar** [lllyasviel/Fooocus#2622] [Q&A]
Everytime I try to do inpainting, the inpainted areas are oversharpened to the point of generating artifacts. Here is an example where the character was inpainted.

![2024-03-24_13-53-29_9822](https://github.com/lllyasviel/Fooocus/assets/329857/5a4f6c22-5063-42b3-97bb-b8e0678e471e)

![2024-03-24_13-55-45_3489](https://github.com/lllyasviel/Fooocus/assets/329857/77358f8c-a29d-4240-87f6-9c75678e65bb)

Both the original and the inpaint where generated using the same parameters, but no matter what, results are always a downgrade and the artifacts become patterns or discolorations. It looks like it tries to add too much detail.

Now, I've read lllyasviel saying something to the effect of not wanting to work on a feature like regional prompter because inpainting could (in theory) do the same, but I believe this is where inpainting will always fall short: it depends on the model being used, right? It seems to me that my issue stems from the fact that this particular model is unable to handle inpainting correctly, and this is precisely where the only solution that comes to mind is to generate the correct picture from the start. Something regional prompter would help doing. 

But I digress. Is this an issue with the inpainting engine or the model?

**Comment by xunkar** [comment]
To add a bit more details: I've tried setting sharpness to 0 and guidance to 2, but it has no effect. Take a look at this one, I think it better captures the issue.

![2024-03-24_14-15-21_8803](https://github.com/lllyasviel/Fooocus/assets/329857/7546a1bb-14a9-40b0-986d-8453755215ef)

**Comment by mashb1t** [comment]
Please provide more background information about thr models you're using, resolution of the target image, etc., best by uploading the full terminal log here.

**Comment by xunkar** [comment]
Well none of the settings affect the result. The target resolution? It's an inpainting, it's the resolution of the base picture (the first one) I don't think the inpainting resolution can be controlled?

The first model is FantasyBlendXL. The civitai link is NSFW so I won't post it here, but the model handles pretty much everything and gives great results in the anime style, which is why I've been using it. However this also happens with much more standard models, like juggernautXL_v8

![2024-03-25_11-48-55_4126](https://github.com/lllyasviel/Fooocus/assets/329857/36d3682a-3182-458a-a615-ab776bbf8145)

![2024-03-25_11-50-05_4575](https://github.com/lllyasviel/Fooocus/assets/329857/58c7f95e-bf25-4dfe-bdfb-c8daf7c8b8ad)

Notice how the inpainted areas are over contrasted and saturated. The prompt for all the pictures was `1girl, fighting, martial arts, dynamic, street art` with style `Graffiti Art` enabled, that's it. And it's all default settings.

Sometimes it seems to be working well until the mid point where the Refiner Swapped stepped happens (internal, I have no refiner enabled).

**Comment by xunkar** [comment]
Well there seems to be something definitely bugged with the inpainting engine.

![image](https://github.com/lllyasviel/Fooocus/assets/329857/cfd91ef4-7bb7-4fc6-9448-5f3d29b6a8ff)

On the left is the source image. The face was masked in inpainting, using "improve details" with "highly detailed face" as an additional prompt. The middle image is the result with the inpainting engine option set on v1, v2.5 or v2.6 (the results are slightly different for each but they all show the same kind of… overwork), the image on the right is using "none" as inpainting engine.
